packages/graph-engine/src/nodes/string/split.ts

```diff
@@ -0,0 +1,41 @@
+import { NodeDefinition, NodeTypes } from "../../types.js";
+
+export const type = NodeTypes.SPLIT_STRING;
+
+export type State = {
+  string: string;
+  separator: string;
+};
+
+export const defaults: State = {
+  string: "",
+  separator: ",",
+};
+
+export const process = (input, state: State) => {
+    const final = {
+        ...state,
+        ...input,
+    };
+
+    if (final.separator === '') {
```

Comment:
Why have a special case for this? final.string.split('') accomplishes the same thing?